src/compiler/crystal/types.cr

```diff
@@ -1047,12 +1047,14 @@ module Crystal
 
     def initialize(program, namespace, name, @superclass, add_subclass = true)
       super(program, namespace, name)
+      superclass = @superclass
```

Comment:
It might be a style issue, but I think it makes more sense the other way around: `@superclass = superclass` (and `superclass` as arg). That's how it worked implicitly before and prevents surprises caused by mismatching types between argument and ivar.